spring-integration-jms/src/test/java/org/springframework/integration/jms/JmsOutboundGatewayTests.java

```diff
@@ -154,4 +181,104 @@ public Message answer(InvocationOnMock invocation) throws Throwable {
 			gateway.stop();
 		}
 	}
+
+	@Test
+	public void testConnectionBreakOnReplyMessageIdCorrelation() throws Exception {
+		CachingConnectionFactory connectionFactory1 = new CachingConnectionFactory(
+				new ActiveMQConnectionFactory("tcp://10.0.0.3:61616"));
```

Comment:
@garyrussell , I don't know how have you tested it. (Looks like you have some external ActiveMQ Broker).
Anyway these tests (with external URL) don't pass even for me, not only for Travis :smile: 
From other side this `61616` fixed port is dangerous.
This one works for me: https://github.com/artembilan/spring-integration/commit/f2840a92a940084dde7650e03829185ad2da0317